core/eolearn/core/utils/parsing.py

```diff
@@ -290,20 +321,22 @@ def get_renamed_features(
 
         If `eopatch` is not provided the method fails if an all-feature request is in the specification.
         """
-        feature_names = []
-        for feature_type, old_name, new_name in self._feature_specs:
-            self._validate_parsing_request(feature_type, old_name, eopatch)
-            if old_name is None and feature_type.has_dict():
-                feature_names.extend(
-                    list(zip(repeat(feature_type), eopatch[feature_type], eopatch[feature_type]))  # type: ignore
-                )
+        feature_names: List[Union[Tuple[FeatureType, str, str], Tuple[FeatureType, None, None]]] = []
```

Comment:
This one is also defined quite a few times. Perhaps it could be extracted as `LongFeatureSpec`

The above is only 50 % a joke